docs/src/dev/provider/index.asciidoc

```diff
@@ -54,6 +54,168 @@ This document attempts to address the needs of the different providers that have
+accept-encoding: deflate
+accept: application/vnd.graphbinary-v4.0
+user-agent: NotAvailable Gremlin-Java.4.0.0 11.0.25 Windows_11.10.0 amd64
+{
+    "gremlin": "g.V()",
+    "language": "gremlin-lang"
+}
+----
+
+==== Expected Request HTTP Headers
+
+[width="100%",cols="3,10,3,3",options="header"]
+|=========================================================
+|Name |Description |Required |Default
+|Accept |Serializer MIME types supported for the response. Must be a mimetype (see table below). |No |`application/vnd.gremlin-v4.0+json;types=false`
+|Accept-Encoding |The requested compression algorithm of the response. Valid values: `deflate`. |No |N/A
+|Authorization |Header used with Basic authorization. |No |N/A
+|Content-Length |The size of the payload |Yes |N/A
+|Content-Type |The MIME type of the serialized body |No |None
+|Gremlin-Hints |A semi-colon separated list of key/value pair metadata that could be helpful to the server in processing a particular request in some way. Must be a hints (see table below). |No |N/A
+|User-Agent |The user agent. Follow the format specified by <<user-agent-format, user agent format>>. |No |<<user-agent-format, user agent format>>
+|=========================================================
+
+==== Request Header Value Options
+
+[width="100%",cols="3,10",options="header"]
+|=========================================================
+|Name |Options
+|mimetype | application/vnd.gremlin-v4.0+json;types=false - GraphSON without embedded types
+|         | application/vnd.gremlin-v4.0+json;types=true - GraphSON with embedded types
+|         | application/vnd.graphbinary-v4.0 - GraphBinary with embedded types
+|hints | mutations: yes, no, unknown - Indicates if the Gremlin contains steps that can mutate the graph.
+|=========================================================
+
+The body of the request should be a `RequestMessage` which is a `Map`. The `RequestMessage` should be serialized using
+the serializer specified by the `Content-Type` header. The following are the key value pairs allowed in a
+`RequestMessage`:
+
+==== Request Message Format
+
+[width="100%",cols="3,10,3,3",options="header"]
+|=========================================================
+|Key |Description |Value |Required
+|gremlin |The Gremlin query to execute. |String containing script |Yes
+|timeoutMs |The maximum time a query is allowed to execute in milliseconds. |Number between 0 and 2^31-1 |No
+|bindings |Any bindings used to execute the query. |Object (Map) |No
+|g |The name of the graph traversal source to which the query applies. Default: "g" |String containing traversal source name |No
+|language |The name of the ScriptEngine to use to parse the gremlin query. Default: "gremlin-lang" |String containing ScriptEngine name |No
+|materializeProperties |Whether to include all properties for results. One of "tokens" or "all". |String |No
+|bulked |Whether the result should be "bulked" (only applies to GraphBinary) |Boolean |No
```

Comment:
could you clarify "bulked" a bit? or link to their meaning if described elsewhere?